src/fstab-generator/fstab-generator.c

```diff
@@ -574,8 +574,33 @@ static int parse_fstab(bool initrd) {
                                 if (streq(canonical_where, where))
                                         canonical_where = mfree(canonical_where);
                                 else
-                                        log_debug("Canonicalized what=%s where=%s to %s",
-                                                  what, where, canonical_where);
+                                        log_debug("Canonicalized where=%s to %s (for what=%s)",
+                                                  where, canonical_where, what);
+                        }
+                }
+
+                if (fstab_test_option(me->mnt_opts, "bind\0" "rbind\0") ||
+                    STR_IN_SET(me->mnt_type, "bind", "rbind")) {
```

Comment:
i figure a helper call would be nice here: `bool mntent_is_bind(const struct mntent *me)` or so, which we could use here... The check is not entirely trivial anymore after all, and this would make things a lot more readable